lib/stat_tracker.rb

```diff
@@ -131,6 +131,36 @@ def total_games_per_team_for(season_id)
     end
   end
 
+  def games_for_team(team_id)
+    @games.select {|game| game.away_team_id == team_id || game.home_team_id == team_id}
+  end
+
+  def opponent_by_game_id_for_team(team_id)
+    games_for_team(team_id).reduce({}) do |result, game|
+      result[game.game_id] =
+        if game.away_team_id == team_id 
+          game.home_team_id 
+        else 
+          game.away_team_id
+        end 
+      result
+    end
+  end
```

Comment:
We could shorten this by replacing the if statements with the following:

`[game.away_team_id, game.home_team_id].select {|id| id != team_id}.join`

But only if you feel like this is more readable!

Reply:
I like this, thank you! I didn't think about using select here instead of if statements. Updated.